hack/build-image/Dockerfile

```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM --platform=linux/amd64 golang:1.19-bullseye
+FROM --platform=linux/amd64 golang:1.20-bullseye
```

Comment:
Any chance you can cherrypick changes from #6085 into this PR?

Reply:
or I guess merge that then rebase this.

Reply:
I prefer to keep the changes in separate PRs.